src/app/pages/send/send-crypto-asset-form/components/recipient-type-dropdown/recipient-type-dropdown.tsx

```diff
@@ -1,36 +1,59 @@
-import { RecipientDropdownItem } from './components/recipient-dropdown-item';
-import { RecipientDropdownLayout } from './components/recipient-dropdown.layout';
+import type { Entries } from '@shared/utils/type-utils';
 
-export enum RecipientFieldType {
-  Address,
-  BnsName,
-}
+import { DropdownMenu } from '@app/ui/components/dowpdown-menu/dropdown-menu';
+import { Flag } from '@app/ui/components/flag/flag';
+import { ChevronDownIcon } from '@app/ui/icons';
+
+import {
+  type RecipientIdentifierType,
+  recipientIdentifierTypesMap,
+} from '../recipient-fields/hooks/use-recipient-select-fields';
 
-interface RecipientTypeDropdownProps {
-  isVisible: boolean;
-  onSelectItem(index: number): void;
-  onSetIsSelectVisible(value: boolean): void;
-  selectedItem: number;
+function makeIteratbleListOfRecipientIdentifierTypes(
+  recipientTypeMap: typeof recipientIdentifierTypesMap
+) {
+  return (Object.entries(recipientTypeMap) as Entries<typeof recipientTypeMap>).map(
+    ([key, value]) => ({ key, label: value })
+  );
 }
-export function RecipientTypeDropdown(props: RecipientTypeDropdownProps) {
-  const { isVisible, onSelectItem, onSetIsSelectVisible, selectedItem } = props;
+const recipientIdentifierTypes = makeIteratbleListOfRecipientIdentifierTypes(
+  recipientIdentifierTypesMap
+);
```

Comment:
maybe put this logic where `recipientTypeMap` actually is?

Reply:
Happy to move if you think that's best. I put it here, because that's the only place it's used